src/org/labkey/test/components/labkey/ui/samples/EntityTypeDesigner.java

```diff
@@ -0,0 +1,156 @@
+package org.labkey.test.components.labkey.ui.samples;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.components.domain.DomainFormPanel;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.params.FieldDefinition;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.labkey.test.WebDriverWrapper.sleep;
+
+public abstract class EntityTypeDesigner extends WebDriverComponent<EntityTypeDesigner.ElementCache>
```

Comment:
Add a type parameter to this so that the subclasses can chain properly.
`sampleTypeDesigner.setName(name).addParentAlias("", "")`  wouldn't work because `getName` doesn't return a SampleTypeDesigner.
```suggestion
public abstract class EntityTypeDesigner<T extends EntityTypeDesigner> extends WebDriverComponent<EntityTypeDesigner.ElementCache>
```
Then add:
```
protected abstract T getThis();
```
And change setters to look like:
```
    public T setName(String name)
    {
        elementCache().nameInput.set(name);
        return getThis();
    }
```

Reply:
Great. Thanks, I'll make that change.